JavaScript/chapter02/p01_remove_dups/avc278.js

```diff
@@ -0,0 +1,101 @@
+// Remove Dups: Write code to remove duplicates from an unsorted linked list.
+// Follow Up: How would you solve this problem if a temporary buffer is not allowed?
+
+const assert = require("assert");
+
+class LinkedListNode {
+  constructor(val, next) {
+    this.val = val === undefined ? 0 : val;
```

Comment:
it's probably better to leave the `val` unchanged: normally containers are thought to be generic data structures that can store anything without imposing any additional restrictions.

in other languages, any kind of restrictions on the values are handled by the type system (by making the containers generic and then enforcing that the values are of a certain type), but JavaScript does not have this luxury :(